### ISSUE TITLE
searching on tags does not find matches in the parent tag name

### ISSUE THREAD

**Issue by AlanBell** [odoo/odoo#24008]
Impacted versions:

11 and earlier

Steps to reproduce:

In the default data on runbot there is a vendor Camptocamp which is tagged as "Partner / Gold"
in Purchases/Vendor view search the tag field for "Partner" and try also searching for "Gold"

Current behavior:

"Partner" finds nothing, "Gold" finds the company.

Expected behavior:

Both should work. Searching for "Partner" should find "Partner/Gold" as well as "Partner/Silver" etc.

**Comment by pedrobaeza** [comment]
I see this more like an improvement than a real bug. It's just that labels are only search by their label name, not the complete path name.

**Comment by AlanBell** [comment]
Arguably so, but it came to me to investigate from a user saying that it wasn't working as expected. It also affects the HR module https://www.odoo.com/forum/help-1/question/how-can-i-search-hr-employee-via-parent-category-10097 from a user perspective, they are searching for some text that is in the tag and it isn't working.

It is a bit hard to see what the point is of having hierarchical tags if you can't group together the things at the parent level. It would be better not to use the feature and in the vendor example tag them as "Partner" and separately tag them as "Gold".

**Comment by pedrobaeza** [comment]
Yeah, let's see if someone from Odoo team can say something about the roadmap in this.

**Comment by mart-e** [comment]
It is actually easy to do, something like odoo-dev/odoo@9ecc60396ff
Now I am wondering on the performance impact.
cc @odony what do you think?

PS: I am not sure to understand the problem with employee categories, there is no hierarchy in [hr.employee.category](https://github.com/odoo/odoo/blob/3f975f97693dfe046dba0d88aaf2bbf2bafa62b6/addons/hr/models/hr.py#L16)

**Comment by odony** [comment]
@mart-e that patch does not work in terms of limits, plus I don't think the spec is correct. What we typically want is to have name_search() symmetrical with name_get(). In the case of hierarchies like tags or stock locations, this requires storing the `complete_name` or simulating it fully when searching. Storing it and searching on the stored version works, but it breaks translations.
So either we find a proper solution for both issues, or we keep it as it is ;-)

**Comment by AlanBell** [comment]
@mart-e yeah, looks like "Parent Employee Tag" was dropped from the HR employee category between version 8 and 9 https://github.com/odoo/odoo/blob/8.0/addons/hr/hr.py#L56 so that doesn't matter now.

**Comment by mart-e** [comment]
I was thinking about the scenario where you filter partners based on the tag, not the CSV import.

To do what you are looking for, the easiest is then probably to add a filter on your search view
```xml
<field name="category_id" operator="child_of"/>
```

**Comment by C3POdoo** [comment]
Dear @AlanBell,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
